exchanges/poloniex/poloniex_futures.go

```diff
@@ -0,0 +1,768 @@
+package poloniex
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchange/order/limits"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/margin"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+)
+
+var (
+	errInvalidPositionMode     = errors.New("invalid position mode")
+	errInvalidMarginAdjustType = errors.New("invalid margin adjust type")
+)
+
+const (
+	tradePathV3    = "/v3/trade/"
+	marketsPathV3  = "/v3/market/"
+	accountPathV3  = "/v3/account/"
+	positionPathV3 = "/v3/position/"
+)
+
+// GetAccountBalance get information about your Futures account.
+func (e *Exchange) GetAccountBalance(ctx context.Context) (*FuturesAccountBalance, error) {
+	var resp *FuturesAccountBalance
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, fGetAccountBalanceEPL, http.MethodGet, accountPathV3+"balance", nil, nil, &resp)
+}
+
+// GetAccountBills retrieve the account’s bills.
+func (e *Exchange) GetAccountBills(ctx context.Context, startTime, endTime time.Time, offset, limit uint64, direction, billType string) ([]*BillDetails, error) {
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if err := common.StartEndTimeCheck(startTime, endTime); err != nil {
+			return nil, err
+		}
+	}
+	params := url.Values{}
+	if !startTime.IsZero() {
+		params.Set("sTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+	}
+	if !endTime.IsZero() {
+		params.Set("eTime", strconv.FormatInt(endTime.UnixMilli(), 10))
+	}
+	if offset > 0 {
+		params.Set("from", strconv.FormatUint(offset, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	if direction != "" {
+		params.Set("direct", direction)
+	}
+	if billType != "" {
+		params.Set("type", billType)
+	}
+	var resp []*BillDetails
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, fGetBillsDetailsEPL, http.MethodGet, accountPathV3+"bills", params, nil, &resp)
+}
+
+// PlaceFuturesOrder place an order in futures trading.
+func (e *Exchange) PlaceFuturesOrder(ctx context.Context, arg *FuturesOrderRequest) (*FuturesOrderIDResponse, error) {
+	if arg.Symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	if arg.Side == "" {
+		return nil, order.ErrSideIsInvalid
+	}
+	if arg.MarginMode != marginMode(margin.Unset) && arg.PositionSide == order.UnknownSide {
+		return nil, fmt.Errorf("%w: either both margin mode and position side filds are filled or left blank", order.ErrSideIsInvalid)
```

Comment:
This only checks one direction. What if PositionSide != unknown and marginMode is unset?